app/models/notification_recipient.rb

```diff
@@ -7,6 +7,7 @@ class NotificationRecipient < ActiveRecord::Base
   validates :user, :presence => true
 
   scope :unseen, -> { where(:seen => false) }
+  after_save :delete_user_cache
```

Comment:
thanks - you are correct, I need to invalidate the cache before the notification itself should expire - will fix.